### PR TITLE
RFC 7159 JSON Using Internal Number 

### PR DESCRIPTION
**Pull request by antonmes**
https://github.com/open-swift/C7/pull/32 afterparty 

Initializers allow writing this code:

``` swift
for item in json {
  guard case .number(let number) = item else {
    throw Error()
  }
  let double = Double(number)
  // do things 
}
```

instead of 

``` swift
for item in json {
  switch item {
  case .number(.double(let value)): // do things
  case .number(.integer(let value)): // do things
  case .number(.unsignedInteger(let value)): // do things
  default: throw Error()
  }
}
```

we should recommend using initializers to avoid this bug, when `1.0` double value send 
from the client-side becomes Integer, example in javascript: `JSON.stringify({num: 1.0})`

### CODE REVIEW

**Comment by czechboy0** [comment]
> #32 afterparty

Be careful what you wish for.

File: Sources/JSON.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+public enum JSON {
+    public enum Number {
+        case integer(Int)
+        case unsignedInteger(UInt)
```
Comment (czechboy0) [inline Sources/JSON.swift:4]: Actually, as far as I can tell we don't have a way to discern between Int and UInt during parsing :/ I'd either just keep (Signed) Int and Double or move to @paulofaria's C7 Number type. Either one is fine with me.

Comment (paulofaria) [inline Sources/JSON.swift:4]: It's possible. If the number starts with `-` it is stored as signed. If not it is stored as unsigned.

Comment (antonmes) [inline Sources/JSON.swift:4]: We should implement it then, because we could have UInt.max

Comment (czechboy0) [inline Sources/JSON.swift:4]: Ha, very good point, I haven't considered that. Yup then it makes sense to have this here 👍

**Comment by paulofaria** [comment]
Related to https://github.com/open-swift/docs/issues/9

**Comment by tanner0101** [comment]
This implementation of JSON has the most support with 6 votes as opposed to <= 3 votes for the other implementations.

**Pull request merged**